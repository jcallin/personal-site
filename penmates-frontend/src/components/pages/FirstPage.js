import React from "react";
import IdealImage from "react-ideal-image"
import landing from '../../media/landing.jpg'
import Tiles from '../Tiles'

export default () => {
    return (
        <div className='landing-image-container'>
            <IdealImage
                className='landing'
                placeholder={{ lqip: landing.preSrc }}
                width={3500}
                height={2095}
                srcSet={[
                    { width: 1000, src: 'media/landing.jpg' },
                    { width: 1000, src: 'media/landing.webp' },
                ]} />
            <Tiles />
            {/* <h2 className='main-title'>Julian Callin</h2> */}
        </div>
    )
}
