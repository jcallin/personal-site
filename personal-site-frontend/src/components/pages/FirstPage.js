import React from "react";
import IdealImage from "react-ideal-image";
import landing from "../../media/landing.jpg";
import Tiles from "../Tiles";

const theme = {
  placeholder: {
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat"
  },
  img: {
    width: "100%",
    height: "100vh",
    objectFit: "cover"
  }
};

export default () => {
  return (
    <div className="landing-container">
      {/* <h2 className="main-title">Julian Callin</h2> */}
      <IdealImage
        className="landing-image"
        theme={theme}
        placeholder={{ lqip: landing.preSrc }}
        width={1920}
        height={1080}
        srcSet={[
          { width: 1000, src: "./media/landing.jpg" },
          { width: 1000, src: "./media/landing.webp" }
        ]}
      />
      <Tiles />
    </div>
  );
};
