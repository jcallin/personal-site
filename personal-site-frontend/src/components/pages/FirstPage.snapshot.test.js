import React from 'react'
import { shallow } from 'enzyme'
import FirstPage from './FirstPage'
describe('First page component', () => {
  it('should have a matching snapshot', () => {
    const wrapper = shallow(<FirstPage />)
    expect(wrapper).toMatchSnapshot()
  })
})
