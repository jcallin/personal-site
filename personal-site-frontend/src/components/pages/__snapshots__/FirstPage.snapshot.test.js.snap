// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First page component should have a matching snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </p>,
      "className": "component first-component",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
      },
      "ref": null,
      "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </p>,
        "className": "component first-component",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
